src/networks/evm/client.ts

```diff
@@ -250,69 +319,42 @@ export class EthereumClient extends Client {
       data: encodedTx,
     };
 
-    const gas = await this.web3.eth.estimateGas(txObject);
-    const gasPrice = Number(await this.web3.eth.getGasPrice());
-    const nonce = await this.web3.eth.getTransactionCount(address);
-    txObject.gas = gas;
-    txObject.gasPrice = `0x${BigInt(Math.ceil(gasPrice * this.gasMultiplier)).toString(16)}`;
+    const txFee = await this.estimateTxFee(txObject);
+    const nonce = await this.getNativeNonce();
+    txObject.gas = txFee.gas.toString();
+    txObject.gasPrice = `0x${BigInt(Math.ceil(Number(txFee.gasPrice) * this.gasMultiplier)).toString(16)}`;
```

Comment:
Same here.

Btw there are a lot of methods that share this part:
```
const txFee = await this.estimateTxFee(txObject);
    const nonce = await this.getNativeNonce();
    txObject.gas = txFee.gas.toString();
    txObject.gasPrice = `0x${BigInt(Math.ceil(Number(txFee.gasPrice) * this.gasMultiplier)).toString(16)}`;
    txObject.nonce = nonce;

    const signedTx = await this.web3.eth.signTransaction(txObject);
    const receipt = await this.sendSignedTx(signedTx.raw);
    return receipt.transactionHash;
```
Does it make sense to move it to a separate function?

Reply:
Thanks for that point. It's really better approach. Moved in https://github.com/zkBob/zeropool-support-js/pull/9/commits/96c6edc93af0da09efec3308f6df99d4df414046